service/storage_service.cc

```diff
@@ -1835,15 +1836,104 @@ future<> storage_service::decommission() {
             }
 
             slogger.info("DECOMMISSIONING: starts");
-            ss.start_leaving().get();
-            // FIXME: long timeout = Math.max(RING_DELAY, BatchlogManager.instance.getBatchlogTimeout());
-            auto timeout = ss.get_ring_delay();
-            ss.set_mode(mode::LEAVING, format("sleeping {} ms for batch processing and pending range setup", timeout.count()), true);
-            sleep_abortable(timeout, ss._abort_source).get();
```

Comment:
What if the user did not enable repair_based_node_operations? Or if the cluster is not fully upgraded?

Reply:
Ah, I see you handle non-upgraded clusters below.

Reply:
Yes. Btw, I plan to move stream basded node operations back to removenode operations if repair_based_node_operations  is disabled. Currently, removenode only uses repair based node ops.